src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -765,25 +780,36 @@ CHIP_ERROR Encode(chip::TLV::TLVWriter & writer, chip::TLV::Tag tag) const
         OnErrorCallbackType aOnError, OnDoneCallbackType aOnDone,
         OnSubscriptionEstablishedCallbackType aOnSubscriptionEstablished = nullptr,
         OnDeviceResubscriptionScheduledCallbackType aOnDeviceResubscriptionScheduled = nullptr)
-        : mAttributePathParamsList(aAttributePathParamsList)
-        , mAttributePathParamsSize(aAttributePathParamsSize)
-        , mEventPathParamsList(aEventPathParamsList)
-        , mEventPathParamsSize(aEventPathParamsSize)
-        , mOnAttributeSuccess(aOnAttributeSuccess)
+        : mOnAttributeSuccess(aOnAttributeSuccess)
         , mOnEventSuccess(aOnEventSuccess)
         , mOnError(aOnError)
         , mOnDone(aOnDone)
         , mOnSubscriptionEstablished(aOnSubscriptionEstablished)
         , mOnDeviceResubscriptionScheduled(aOnDeviceResubscriptionScheduled)
         , mBufferedReadAdapter(*this)
     {
+        if (aAttributePathParamsList != nullptr) {
+            mAttributePathParamsList
+                = Platform::MakeMemoryUnique<app::AttributePathParams>(aAttributePathParamsSize, sizeof(AttributePathParams));
```

Comment:
This is not great, because this allocation can fail...

I think the previous setup, where the callback just held pointers to these things and something else handled the memory management was better than making these copies inside the callback....

Reply:
I agree your opinion. I will change this code.